src/app.ts

```diff
@@ -172,6 +236,51 @@ function registerRoutes(app: Express): void {
   );
 }
 
+/**
+ * Initializes SpidStrategy for passport and setup /login route.
+ */
+function registerLoginRoute(
+  app: Express,
+  newSpidStrategy: passport.Strategy
+): void {
+  // Add the strategy to authenticate the proxy to SPID.
+  passport.use("spid", newSpidStrategy);
+  const spidAuth = passport.authenticate("spid", { session: false });
+  app.get("/login", spidAuth);
+
+  app.get("/metadata", (_0, res) => {
+    const metadata = ((newSpidStrategy as unknown) as SpidStrategy).generateServiceProviderMetadata(
```

Comment:
why the cast?

Reply:
btw generating the metadata is quite expensive, we should generate it once, not on each request

Reply:
Indeed, I checked again the code and there was a mistake in the typization of `newSpidStrategy`, because its type is already `SpidStrategy`. A cast is needed anyway in [line 247](https://github.com/teamdigitale/io-onboarding-pa/pull/9/files/fc157d763b95a31b22251c1079db6fd9b8abb595#diff-f98485da1849ae631ba5c04e5cec07ddR247) because `passport.use()` requires a parameter of type `Strategy`.